pkg/apis/stork/v1alpha1/applicationclone.go

```diff
@@ -0,0 +1,86 @@
+package v1alpha1
+
+import (
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	//ApplicationCloneResourceName is the name for the application clone resource
+	ApplicationCloneResourceName = "applicationclone"
+	//ApplicationCloneResourcePlural is the name in plural for the application clone resources
+	ApplicationCloneResourcePlural = "applicationclones"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+//ApplicationClone represents the cloning of application in different namespaces
+type ApplicationClone struct {
+	meta.TypeMeta   `json:",inline"`
+	meta.ObjectMeta `json:"metadata,omitempty"`
+	Spec            ApplicationCloneSpec   `json:"spec"`
+	Status          ApplicationCloneStatus `json:"status,omitempty"`
+}
+
+//ApplicationCloneSpec defines the spec to create an application clone
+type ApplicationCloneSpec struct {
+	//NamespaceMapping to store the target:destination namespaces
+	NamespaceMapping map[string]string `json:"namespace_mapping"`
+	//Selectors for label on objects
+	Selectors    map[string]string `json:"selectors"`
+	PreExecRule  string            `json:"preExecRule"`
```

Comment:
We need to add a `ReplacePolicy` here to govern what happens if there are volumes/resources with the same name in the destination namespace. ReplacePolicy can be `Delete`,`Retain`, `Fail`